lib/topics_filter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class TopicsFilter
+  def self.register_filter(matcher, &block)
+    self.filters[matcher] = block
+  end
+
+  def self.filters
+    @@filters ||= {}
+  end
+
+  register_filter(/\Astatus:([a-zA-Z]+)\z/i) do |topics, match|
+    case match
+    when "open"
+      topics.where("NOT topics.closed AND NOT topics.archived")
+    when "closed"
+      topics.where("topics.closed")
+    when "archived"
+      topics.where("topics.archived")
+    when "deleted"
+      if @guardian.can_see_deleted_topics?(@category)
+        topics.unscope(where: :deleted_at).where("topics.deleted_at IS NOT NULL")
+      else
+        topics
+      end
+    end
+  end
+
+  def initialize(guardian:, scope: Topic, category_id: nil)
+    @guardian = guardian
+    @scope = scope
+    @category = category_id.present? ? Category.find_by(id: category_id) : nil
+  end
+
+  def filter(input)
+    input
+      .to_s
+      .scan(/(([^" \t\n\x0B\f\r]+)?(("[^"]+")?))/)
```

Comment:
This regexp is lifted from https://github.com/discourse/discourse/blob/3dbaaf1268754f1e57de0b28b2995f1e462f6232/lib/search.rb#L774. I didn't find any tests regarding this but better to err on the safer side for now and use the same pattern.